packages/pcd/gpc-pcd/src/GPCPCDPackage.ts

```diff
@@ -268,42 +280,71 @@ function validateInputPOD(
   podPCD: PODPCD,
   params: PODPCDArgValidatorParams | undefined
 ): boolean {
-  if (podPCD.type !== PODPCDTypeName) {
-    return false;
+  // Bypass checks if there is nothing to work with.
+  if (params === undefined) {
+    return true;
   }
 
-  if (params?.proofConfig !== undefined) {
-    let proofConfig: GPCProofConfig;
-    try {
-      proofConfig = deserializeGPCProofConfig(params.proofConfig);
-    } catch (e) {
-      if (e instanceof TypeError) {
-        params.notFoundMessage = e.message;
-        return false;
-      }
-      throw e;
-    }
+  // Deserialise provided parameters
+  let proofConfig: GPCProofConfig | undefined;
+  let membershipLists: PODMembershipLists | undefined;
+  let prescribedEntries: PODEntryRecord | undefined;
+
+  try {
+    proofConfig =
+      params.proofConfig !== undefined
+        ? deserializeGPCProofConfig(params.proofConfig)
+        : undefined;
+
+    membershipLists =
+      params.membershipLists !== undefined
+        ? podMembershipListsFromSimplifiedJSON(params.membershipLists)
+        : undefined;
 
-    // POD podName should be present in the config and have all
-    // entries specified there.
-    const podConfig = proofConfig.pods[podName];
-    if (podConfig === undefined) {
-      params.notFoundMessage = `The proof configuration does not contain this POD.`;
+    prescribedEntries =
+      params.prescribedEntries !== undefined
+        ? podEntryRecordFromSimplifiedJSON(params.prescribedEntries)
+        : undefined;
+  } catch (e) {
+    if (e instanceof TypeError || e instanceof Error) {
+      params.notFoundMessage = e.message;
       return false;
-    } else {
-      const entries = Object.keys(podConfig.entries);
-      // Enumerate POD entries
-      const podEntries = podPCD.pod.content.asEntries();
-      // Return true iff all elements of `entries` are keys of `podEntries`
-      return entries.every((entryName) => podEntries[entryName] !== undefined);
     }
+    throw e;
   }
 
+  const prescribedSignerPublicKeys = params.prescribedSignerPublicKeys;
+
+  return (
```

Comment:
I'd suggest ordering these by likely cost, so that short-circuit evaluation is most likely to give fast answers.  In particular I think lists can get large which would make that check more expensive than prescribed values, so maybe put it later.